pkg/sources/adapter/salesforcesource/auth/fake/authenticator.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fake
+
+import (
+	"errors"
+
+	"github.com/triggermesh/triggermesh/pkg/sources/adapter/salesforcesource/auth"
+)
+
+// Authenticator is a test oriented fake Authenticator.
+type Authenticator struct {
+	defaultCredentials auth.Credentials
+}
+
+var _ auth.Authenticator = (*Authenticator)(nil)
+
+// NewFakeAuthenticator creates a Fake authenticator for Bayeux/Salesforce.
+func NewFakeAuthenticator(defaultCredentials auth.Credentials) auth.Authenticator {
```

Comment:
Does it even need to receive credentials? Couldn't it just always return the same (fake) credentials since we are in the context of a test?

Reply:
The credentials contain the URL to connect to, I needed to inject the credentials into the fake authenticator to use the `httptest.Server` URL.

We could re-visit this if we find a different case to maybe add an argument-less `NewFakeAuthenticator()`.
As an alternative we could use the `NewFakeAuthenticator(opts ...FakeAuthenticatorOptions)` and I could be using the option to inject my own credentials from the test.
Thoughts?

Reply:
I just wanted to ensure the parameter was being used. The current approach looks fine to me 👌